nova/scheduler/filters/shard_filter.py

```diff
@@ -82,9 +84,19 @@ def _update_cache(self):
             data = resp.json()
             for project in data['projects']:
                 project_id = project['id']
-                shards = [t for t in project['tags']
-                          if t.startswith(self._SHARD_PREFIX)]
+                tags = project['tags']
+                shards = [t for t in tags if t.startswith(self._SHARD_PREFIX)]
+                # NOTE(jakob): If the project has more than one shard enabled
+                # in _any_ AZ, consider all shards is enabled in _all_ AZs.
+                # The proper `sharding_enabled` tag is appended if not already
+                # in the actual list of tags.
+                shard_counts = Counter("".join(shard.split("-")[:2])
```

Comment:
What's the point of this split-join? `vc-a-1` becomes `vca`. Is this to accommodate for `[:4]` not catching everything of the tag? Then it might break if we do `vca-1` in the future, too. So I don't think we can safeguard against us changing the structure of the tags.

Reply:
Not relevant anymore, but i just disliked the `[:4]` it looked to magic to me. I thought it's more likely that the length of the fields between the "-"s would change than the use of "-"-separated fields itself.